webpack.config.js

```diff
@@ -1,6 +1,7 @@
 const path = require('path');
 const webpack = require('webpack');
 const CircularDependencyPlugin = require('circular-dependency-plugin');
+const createInnerJsxTransformer = require('typescript-plugin-inner-jsx').default;
```

Comment:
This and other lines would not be necessary if we would just use `ttsc`. But maybe you tried that already.

However, I would prefer if we move this all to a configuration in the tsconfig (which is what `ttsc` allows us to do).

Reply:
`ttsc` won't work in this case. Transform needs to be applied **before** `emit` step, while by using `ttsc` it can be applied only **during** different steps of `emit`.

The transform needs to be applied before emit because otherwise updated declarations are not generated.
`afterDeclarations` can't be used because on this step the input data provided to the transformer contains declarations only and not the full source, so the transformer can't build inner structure from it.

Reply:
Okay understood. But then I think we should not have the build.ts here. Rather, it should be already in the ts transform package, because others will also need this runner then. It should come in form of an executable (i.e., register in bin) such that it may be executed directly (at least from npm scripts / or if installed globally / via npx / ...).

Reply:
Don't agree. I believe that transformer package should contain transformer only and it's integration into the system is up to the consumer. We can only advice how to integrate it.
Transformer itself follows all the principles and does it's job perfectly - transforms code from A to B. Compiler definitely doesn't belong to the package because it serves different purpose. And the reason why we need a custom compiler here because:

- typescript doesn't have standard way to run transformers through tsc arguments or tsconfig
- we need to change order transfomer runs
- we need not only js files but d.ts as well

...And the end consumer may not have these reasons

Reply:
That does not make much sense. We just supply the standard runner - we do not force anyone to use it. However, any implementation should look similar to ours, so not providing this default implementation would be a waste. Even if someone would want to roll out his/her own implementation - it would be not different than now. So

- users who are fine with our implementation don't have to do anything
- other users are not affected

There is no reason to not roll it out in the typescript transform package. Again, it does not force anyone to use it.

Reply:
Just to be clear - what the transformer package does is pretty much what we want from the runner. Of course, we want js in the end. Of course, we want to have the d.ts affected (otherwise its only half of a job). Of course, we want it in the stage you describe.

Again, if someone has something else in mind including it in the package already will not have a negative impact here. There are only advantages.